### PR TITLE
[doc] [engine] Document `abort_on_undefined_evars`.

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

File: dev/doc/changes.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+## Changes between Coq 8.8 and Coq 8.9
+
+### ML API
+
+Proof engine
+
+  Note that the core function `EConstr.to_constr` now _enforces_ by
+  default that the resulting term is ground, that is to say, free of
+  Evars. This is usually what you want, as for the typing system to
+  work, open terms should indeed be of type `EConstr.t`.
```
Comment (SkySkimmer) [inline dev/doc/changes.md]: >for the typing system to work

What does that mean?

Comment (ejgallego) [reply dev/doc/changes.md]: Tweaked.

File: dev/doc/changes.md [hunk 2]
```diff
@@ -1,3 +1,20 @@
...
+
+### ML API
+
+Proof engine
+
+  Note that the core function `EConstr.to_constr` now _enforces_ by
+  default that the resulting term is ground, that is to say, free of
+  Evars. This is usually what you want, open terms should indeed be of
+  type `EConstr.t` as to profit of the invariants the `EConstr` API is
+  meant to provide.
```
Comment (SkySkimmer) [inline dev/doc/changes.md]: I suggest "open terms should be of type EConstr.t to benefit from the invariants the EConstr API guarantees"

Comment (ejgallego) [reply dev/doc/changes.md]: Thanks, incorporated almost all, but kept the "is meant to" as IMHO the softer language matches the actual guarantees better.

**Comment by ejgallego** [comment]
Nobody taking on this, @ppedrot ?

**Review by ppedrot — APPROVED** [review]
Doesn't hurt to merge this kind of documentation.

**Pull request merged**